README.md

```diff
@@ -259,6 +263,10 @@ public List<SomeClass> ListForReturnListFor()
 
 Now we are really getting in weird territory. Arguably, optimizing things at the level of `foreach` vs `for` can be considered to be too crazy and esoteric. Like with all things, it is crucial to know when to stop on a given code path and find other areas that are more impactful to optimize. The context of the piece of code that you are trying to optimize is crucial. For example, if you are trying to optimize something that uses `IEnumerable` that is passed based on the user input like as for the `AmqReceiver` by applying the rules above you might turn this piece of code:
 
+>> it is crucial to know when to stop on a given code path and find other areas that are more impactful to optimize
+>
+> Why not going all-in?
```

Comment:
What I want to say here is that when you are overly focused on one piece of the code for too long you might end up making the situation worse (the example I show) or neglect/forget other parts that you can make bigger impacts in. I guess your question hints that I haven't done a good enough job to convey this message, am I right?

Reply:
> I guess your question hints that I haven't done a good enough job to convey this message, am I right?

Correct. I think it's helpful to provide some criteria to decide when to stop.